packages/back-end/src/validators/event-webhook.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from "zod";
+import { zodNotificationEventNamesEnum } from "../events/base-types";
+
+export const eventWebHookPayloadTypes = ["raw", "slack", "discord"] as const;
+
+export type EventWebHookPayloadType = typeof eventWebHookPayloadTypes[number];
+
+export const eventWebHookMethods = ["POST", "PUT", "PATCH"] as const;
+
+export type EventWebHookMethod = typeof eventWebHookMethods[number];
+
+export const eventWebHookInterface = z
+  .object({
+    id: z.string(),
+    organizationId: z.string(),
+    dateCreated: z.date(),
+    dateUpdated: z.date(),
+    url: z.string().url(),
+    name: z.string().trim().min(2),
+    events: z.array(z.enum(zodNotificationEventNamesEnum)).min(1),
+    enabled: z.boolean(),
+    projects: z.array(z.string()),
+    tags: z.array(z.string()),
+    environments: z.array(z.string()),
+    payloadType: z.enum(eventWebHookPayloadTypes),
+    method: z.enum(eventWebHookMethods),
+    headers: z.object({}).catchall(z.string()),
```

Comment:
Instead of object and `catchall`, should we do this?
```ts
z.record(z.string())
```

Reply:
Definitely.